src/test/java/edu/ucla/library/iiif/fester/handlers/GetCollectionHandlerTest.java

```diff
@@ -0,0 +1,84 @@
+
+package edu.ucla.library.iiif.fester.handlers;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.junit.Test;
+
+import edu.ucla.library.iiif.fester.Config;
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.HTTP;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import edu.ucla.library.iiif.fester.utils.IDUtils;
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.util.StringUtils;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+
+public class GetCollectionHandlerTest extends AbstractFesterHandlerTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetCollectionHandlerTest.class, Constants.MESSAGES);
+
+    private final String myCollectionS3Key = IDUtils.getCollectionS3Key("ark:/21198/zz0009gsq9");
+
+    /**
+     * Test the GetCollectionHandler.
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    @SuppressWarnings("deprecation")
+    public void testGetCollectionHandler(final TestContext aContext) throws IOException {
+        final String expectedCollection = StringUtils.read(COLLECTION_FILE);
+        final Async asyncTask = aContext.async();
+        final int port = aContext.get(Config.HTTP_PORT);
+        final String requestPath = IDUtils.getResourceURIPath(myCollectionS3Key);
+
+        LOGGER.debug(MessageCodes.MFS_137, requestPath);
+
+        myVertx.createHttpClient().getNow(port, Constants.UNSPECIFIED_HOST, requestPath, response -> {
+            final int statusCode = response.statusCode();
+
+            if (response.statusCode() == HTTP.OK) {
+                response.bodyHandler(body -> {
+                    final String foundCollection = body.toString(StandardCharsets.UTF_8);
+
+                    // Verify that the CORS header is permissive
+                    aContext.assertEquals(Constants.STAR, response.getHeader(Constants.CORS_HEADER));
+
+                    // Verify that our retrieved JSON is as we expect it
+                    aContext.assertEquals(new JsonObject(expectedCollection), new JsonObject(foundCollection));
```

Comment:
A `aContext.assertEquals` will complete the task when the assertion fails so in these situations we've been wrapping the `asyncTask.complete()` calls in `if (!asyncTask.isCompleted()) { }` to avoid the warning message that comes out in the log when you complete a task that's already been completed. This also applies to line 81.